grafana-plugin/src/components/ManualAlertGroup/ManualAlertGroup.tsx

```diff
@@ -1,235 +1,73 @@
-                    target="_blank"
-                    rel="noreferrer"
-                    className={cx('link')}
-                  >
-                    <Text type="link">Learn more.</Text>
-                  </a>
-                </Text>
-              </HorizontalGroup>
-            </Alert>
-          ))}
-      </VerticalGroup>
-    );
-  };
+  const { selectedTeamResponder, selectedUserResponders, resetSelectedUsers, resetSelectedTeam } =
+    useContext(DirectPagingContext);
+
+  const hasSelectedEitherATeamOrAUser = selectedTeamResponder !== undefined || selectedUserResponders.length > 0;
+
+  // TODO: add a loading state while we're waiting to hear back from the API when submitting
+  // const [directPagingLoading, setdirectPagingLoading] = useState<boolean>();
+
+  // const [chatOpsAvailableChannels, setChatopsAvailableChannels] = useState<any>();
+
+  const handleFormSubmit = useCallback(
+    async (data: FormData) => {
+      const transformedData = prepareForUpdate(selectedUserResponders, selectedTeamResponder, data);
+
+      const resp = await directPagingStore.createManualAlertRule(transformedData);
+
+      if (!resp) {
+        openWarningNotification('There was an issue creating the alert group, please try again');
+        return;
+      }
+
+      resetSelectedUsers();
+      resetSelectedTeam();
+
+      onCreate(resp.alert_group_id);
+      onHide();
+    },
+    [prepareForUpdate, selectedTeamResponder, selectedUserResponders, resetSelectedUsers, resetSelectedTeam]
+  );
 
   return (
-    <Drawer scrollableContent title="Create Alert Group" onClose={onHide} closeOnMaskClick={false} width="70%">
+    <Drawer scrollableContent title="New escalation" onClose={onHide} closeOnMaskClick={false} width="70%">
       <VerticalGroup>
+        {/* TODO: pass in generic form data here */}
         <GForm form={manualAlertFormConfig} data={data} onSubmit={handleFormSubmit} />
-        <Field label="Team to notify">
-          <GrafanaTeamSelect withoutModal onSelect={onUpdateSelectedTeam} />
-        </Field>
-        <DirectPagingIntegrationVariants
-          selectedTeamId={selectedTeamId}
-          selectedTeamDirectPaging={selectedTeamDirectPaging}
-          chatOpsAvailableChannels={chatOpsAvailableChannels}
-        />
-        <EscalationVariants
-          value={{ userResponders, scheduleResponders }}
-          onUpdateEscalationVariants={onUpdateEscalationVariants}
-          variant={'secondary'}
-          withLabels={true}
-        />
+        <AddResponders mode="create" />
         <HorizontalGroup justify="flex-end">
           <Button variant="secondary" onClick={onHide}>
             Cancel
           </Button>
-          <Button type="submit" form={manualAlertFormConfig.name} disabled={!selectedTeamId}>
+          {/* TODO: make the button be disabled if the form is not valid */}
+          <Button type="submit" form={manualAlertFormConfig.name} disabled={!hasSelectedEitherATeamOrAUser}>
```

Comment:
@grafana/grafana-oncall-frontend I'd like to disable this submit button in the following scenario:
- the form data, managed by the `GForm` component (in this case just the "message" field) is not valid (ie. empty)
AND
- the user hasn't selected _at least_ a team _or_ one user (this state is managed via a React context; see `selectedTeamResponder` and `selectedUserResponders`)

The second condition is easy to disable based on, but I don't see how to "reach into" the `GForm` component to get its current validation state?

Reply:
For the first one I believe there's no support for it at the moment, but you can easily pass the Form data throughout the `onChange` handler if you add a prop for it. Something among these lines

- Add a new optional prop on the `Form` component named `onChange`
- Within `Form` whenever `onChange` is being called, call the `onChange` (if any) prop as well with the needed data (in this case I believe it's just `errors` and possibly `getValues`)
- Inside the `onChange` code you pass to the component you can iterate over the `errors` object and see if there's any key in there, if it is then there's an error and from there you can enable/disable the submit. You can also use the `getValues` I believe if you wanna look for a specific value

Hope that helps.

Reply:
this should be addressed in [commit a21492a](https://github.com/grafana/oncall/pull/3128/commits/a21492ae02573a83187e3aee4fc3934b91b84aa2)